database/database.go

```diff
@@ -140,41 +135,42 @@ func (db *DB) UserByID(ctx context.Context, id string) (*User, error) {
 	return &u, nil
 }
 
-// UserCreate creates a new user in the DB. We need the user object to be passed
-// by reference because we need to be able to update the ID of new user.
-func (db *DB) UserCreate(ctx context.Context, u *User) error {
-	if !u.Email.Validate() {
-		return ErrInvalidEmail
-	}
-	// Check for an existing user with this email.
-	users, err := db.managedUsersByField(ctx, "email", string(u.Email))
+// UserCreate creates a new user in the DB.
+func (db *DB) UserCreate(ctx context.Context, sub string, tier int) (*User, error) {
+	// Check for an existing user with this sub.
+	users, err := db.managedUsersByField(ctx, "sub", sub)
 	if err != nil && !errors.Contains(err, ErrUserNotFound) {
-		return errors.AddContext(err, "failed to query DB")
+		return nil, errors.AddContext(err, "failed to query DB")
 	}
 	if len(users) > 0 {
-		return ErrEmailAlreadyUsed
+		return nil, ErrUserAlreadyExists
+	}
+	u := &User{
+		ID:   primitive.ObjectID{},
+		Sub:  sub,
+		Tier: tier,
 	}
 	// Insert the user.
 	fields, err := bson.Marshal(u)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	ir, err := db.staticUsers.InsertOne(ctx, fields)
 	if err != nil {
-		return errors.AddContext(err, "failed to Insert")
+		return nil, errors.AddContext(err, "failed to Insert")
 	}
 	u.ID = ir.InsertedID.(primitive.ObjectID)
 	// Sanity check because races exist.
-	users, err = db.managedUsersByField(ctx, "email", string(u.Email))
+	users, err = db.managedUsersByField(ctx, "sub", sub)
 	if len(users) > 1 {
 		// Race detected! Email no longer unique in DB. Delete new user.
```

Comment:
I don't think it's a good idea to go and delete users when you found they exist twice. Wouldn't it be possible to end up going from 2 -> 0 users? Also simply deleting a user.. I don't think we should be doing that ever.

Reply:
Perhaps we should consider soft-deletion 🤔  (in general, not in this particular situation)

Reply:
Yeah I don't think we need to worry about this. If we try and create a user with an existing sub it should just fail with an error. This check doesn't really protect us anyway cause things might happen in the database between getting the user and deleting it. So we can just hope that the database enforces unique subs.